libs/libc/stdio/lib_ungetc.c

```diff
@@ -52,7 +53,7 @@ int ungetc(int c, FAR FILE *stream)
 
   /* Stream must be open for read access */
 
-  if ((stream->fs_fd < 0) || ((stream->fs_oflags & O_RDOK) == 0))
+  if ((fd < 0) || ((stream->fs_oflags & O_RDOK) == 0))
```

Comment:
should we allow cookie case? which may pass in the negative value

Reply:
done